crates/core_simd/src/macros.rs

```diff
@@ -0,0 +1,143 @@
+macro_rules! from_aligned {
+    { unsafe $from:ty => $to:ty } => {
+        impl core::convert::From<$from> for $to {
+            #[inline]
+            fn from(value: $from) -> $to {
+                assert_eq!(core::mem::size_of::<$from>(), core::mem::size_of::<$to>());
+                assert!(core::mem::align_of::<$from>() >= core::mem::align_of::<$to>());
+                unsafe { core::mem::transmute(value) }
+            }
+        }
+    };
+    { unsafe $a:ty |bidirectional| $b:ty } => {
+        from_aligned!{ unsafe $a => $b }
+        from_aligned!{ unsafe $b => $a }
+    }
+}
+
+macro_rules! from_unaligned {
+    { unsafe $from:ty => $to:ty } => {
+        impl core::convert::From<$from> for $to {
+            #[inline]
+            fn from(value: $from) -> $to {
+                assert_eq!(core::mem::size_of::<$from>(), core::mem::size_of::<$to>());
+                unsafe { (&value as *const $from as *const $to).read_unaligned() }
+            }
+        }
+    }
+}
```

Comment:
I completely forgot it validates the size already.  I also didn't realize that it would fix the alignment, but that makes sense.  The simplified macros are much simpler.  I also added a macro specifically for converting to x86 types.